scripts/pre_commit_linter.py

```diff
@@ -603,13 +607,13 @@ def _check_bad_patterns(all_files):
 
             if filename.endswith('.js'):
                 for regexp in BAD_PATTERNS_JS_REGEXP:
-                    if _check_bad_pattern_in_file(filename, content, regexp):
+                    if _check_bad_pattern_in_file(filename, regexp):
```

Comment:
Also, now the `_check_bad_pattern_in_file` function does not need `content` argument. So it has been removed.

Reply:
I'm a little bit iffy about this, since you're requiring 2 file reads instead of 1, and file reads take time. Might be worth checking whether they take more or less time than splitting `content` by `\n`.

Reply:
Okay. We can do something like this:  
```python
for line_num, line in enumerate(content.split('\n'), 1):
```  
How does this look?  
Thanks!

Reply:
That's fine with me, but did you time it?

Reply:
OK, thanks! Sounds good then.

Reply:
Got it. Thanks!